### PR TITLE
[Chat][Feature]Message Deleted Redux Service Layer 

### PR DESCRIPTION
**Pull request by ShaunaSong**
## Purpose

* Add messages deleted logic

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Pull Request Type
What kind of change does this Pull Request introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Pull Request Checklist

- [ ] Public API changes
- [ ] Verified for cross-platform
- [ ] Synced with iOS, Web team
- [ ] Internal review completed
- [ ] UI Changes
  - [ ] Screen captures included
  - [ ] Tested for Light/Dark mode
  - [ ] Tested for screen rotation
  - [ ] Tested on Tablet and small screen device (5")
  - [ ] Include localization changes
- [ ] Tests
  - [ ] Memory leak analysis performed
  - [ ] Tested on API 21, 26 and latest 
  - [ ] Tested for background mode
  - [ ] Unit Tests Included
  - [ ] UI Automated Tests Included

## How to Test

* Open...
* Click on...

## What to Check
Verify that the following are valid
* ...

### CODE REVIEW

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/repository/RepositoryMiddleware.kt [hunk 1]
```diff
@@ -44,6 +45,11 @@ internal class RepositoryMiddlewareImpl(
         notifyUpdate(dispatch)
     }
 
+    private fun processDeleteMessage(action: ChatAction.DeleteMessage, dispatch: Dispatch) {
+        messageRepository.deleteMessage(action.messageInfoModel)
+        notifyUpdate(dispatch)
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/repository/RepositoryMiddleware.kt:50]: do we need it here?

File: azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandler.kt [hunk 2]
```diff
@@ -36,6 +37,10 @@ internal class ChatActionHandler(private val chatService: ChatService) {
         chatService.requestPreviousPage()
     }
 
+    private fun deleteMessage(action: ChatAction.DeleteMessage) {
+        chatService.deleteMessage(action.messageInfoModel.id.toString())
```
Comment (iaulakh) [inline azure-communication-ui/chat/src/main/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandler.kt:41]: can we handle success/error here?

**Review by iaulakh — COMMENTED** [review]
please write test cases

**Comment by ShaunaSong** [comment]
> please write test cases

Done, please double check

File: azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandlerUnitTest.kt [hunk 3]
```diff
@@ -112,6 +112,81 @@ internal class ChatActionHandlerUnitTest : ACSBaseTestCoroutine() {
...
+                on { dispatch(any()) } doAnswer { }
+            }
+
+            // act
+            chatHandler.onAction(action, mockAppStore::dispatch)
+
+            deleteChatMessageCompletableFuture.complete(any())
+
+            // assert
+            verify(mockAppStore, times(1)).dispatch(
```
Comment (ahammer) [inline azure-communication-ui/chat/src/test/java/com/azure/android/communication/ui/chat/redux/middleware/sdk/ChatActionHandlerUnitTest.kt:147]: I don't think we need to verify Dispatch. (should be covered with redux tests)

However, we should have action->effect tests. So this one should be verifying ChatService.deleteMessage was called correctly.

There should be another test for the Repository Middleware that is
Action->RepositoryMiddleware.onDelete, similar to this one.

**Review by ahammer — APPROVED** [review]

**Review by Mohtasim — APPROVED** [review]

**Pull request merged**